File: Basic DeNico/basic-denico.php
<?php
/**
 * Created by PhpStorm.
 * User: kovalevcon
 * Date: 28.09.18
 * Time: 18:38
 */

/**
 * Basic DeNico variation function
 *
 * @param string $key
 * @param string $message
 * @return string
 */
function de_nico(string $key, string $message): string
{
    $lengthKey = strlen($key);
    $sortKey = str_split($key);
    sort($sortKey);
    $sortKey = array_flip($sortKey);

    $hash = [];
    for ($i = 0; $i < $lengthKey; $i++) {
        $hash[$key[$i]] = $sortKey[$key[$i]];
    }

    $result = "";
    for ($i = 0, $lengthMsg = strlen($message); $i < $lengthMsg; $i += $lengthKey) {
        foreach (str_split($key) as $x) {
            $temp = mb_substr($message, $i, $i + $lengthKey);
            $result .= $temp[$hash[$x]] ?? ' ';
        }
    }

    return rtrim($result);
}

echo "<pre>de_nico(\"crazy\", \"cseerntiofarmit on  \");\r\n";
echo de_nico("crazy", "cseerntiofarmit on  "); // 'secretinformation'
echo "</pre>";

File: Basic DeNico/basic-denico.py
def de_nico(key, msg):
    """
    :type key: str
    :type msg: str
    :rtype: str
    """
    len_key = len(key)
    keys = dict(zip(key, [int(''.join(sorted(key)).find(x)) for x in key]))

    res = ''
    for i in range(0, len(msg), len_key):
        for x in key:
            temp = msg[i: i + len_key]
            try:
                res += temp[keys[x]]
            except IndexError:
                pass

    return res.rstrip()


print(de_nico("crazy", "cseerntiofarmit on"))  # secretinformation
